aips/aip-6.md

```diff
@@ -0,0 +1,84 @@
+
+## Rationale
+
+This feature has the potential to increase the number of validators in the ecosystem leading to further decentralization and a higher amount of tokens staked on the protocol.
+
+## Specification
+
+A detailed specification of the proposed solution can be found [here](https://docs.google.com/document/d/1wmE_TV3AsYP_lAtSYOZg9tqJZ9_BeRCvfrz7OHXv7zA/edit?usp=sharing).
+
+To summarize, the following behavior would be supported:
+
+- Admin of the delegation pool = node operator
+- Initiates delegation pool
+- Join validator set
+- Leave validator set
+- Set commission rate at the start of contract
+  - Contract will pay commission out first, then rewards to principal
+  - Commission rate cannot be changed
+- Delegators = token owners
+  1.  Add stake
+  2.  Schedule stake for unlocking
+  3.  Cancel unlocking of stake (moves stake to previous state)
+  4.  Withdraw stake
+- Reset-lockup = No one
+- Delegated voter = Admin
+
+## Reference Implementation
+
+There is a reference autonomous (external to the Aptos framework) and tested implementation treating the stake module as a black box at https://github.com/bwarelabs/aptos-delegation-pool.
+Additionally, an example delegation pool, linked to a running validator node and active on the Aptos testnet, can be examined on the [explorer](https://explorer.aptoslabs.com/account/0x61f9e2697d5eee926e990b537a8286a029c908c1ae117e76ddf7a849bf87fb59/resources).
+
+The integration into the Aptos framework would preserve the rewards-distribution formula and the accounting of delegators' stakes, while deprecating resources already available within the framework and responsible for tracking epochs, rewards and pool's aggregated stakes.
+
+The delegation pool would remain a proxy resource for the underlying stake pool, while the factory module could support the wrapping of existing stake pools into delegation ones in the future.
+
+## Risks and Drawbacks
+
+The staking API initially exposed would incur a higher gas cost (only for delegation pools) as additional resources have to be stored and maintained in order to keep track of individual delegators' stakes and cumulative rewards earned by pool's aggregated stakes.
+
+Compared to a single-owner stake pool, rewards produced each epoch would not be automatically restaked for delegators as this would introduce a quadratic complexity when updating the validators set at the epoch's end. Nonetheless, delegators can manually restake their earned rewards, while failing to do so over an entire year would result in their compound APR dropping from 16.18% (automatically restaking each epoch of 1h duration) to 15% (current reward rate).
+
+The operator fee, previously configurable by the owner, could be set either by a governance process at the level of the delegation pool or immutably at pool's creation by the node operator itself. For the latter, delegators have the option to participate in pools of their choice with regard to fee and node performance.
```

Comment:
Which governance process is "governance process at the level of the delegation pool" referring to?

Reply:
This was before deciding that it is straightforward to have an immutable fee and let delegators choose the delegation pool themselves.

Reply:
I think making an immutable fee for a public pool is a kinda controversial idea in the long term. It can potentially limit flexibility for validators to reflect changing market conditions and staking revenues.

For example, changing the fee to 0% can be a good option to compensate for a short-term weak performance/incident that led to a reward loss without the need to send and calculate rebates per user, imposing a legal risk of a custodial transfer. 

In case of increasing the fee, responsible validators make announcements in advance, allowing delegators to change the pool if they don't want to continue staking at that fee. There are also other options to prevent that behavior, like building a notification tool, setting the max cap for possible fee % during the pool creation etc.

Curious if the decision to make an immutable fee is based on tech limitations, or are there other reasons for that?

Reply:
Given the nascency of delegation on aptos blockchain, I would suggest an immutable fee helps to protect the stakers by ensuring that the operator doesn't suddenly raise fees without notice / public announcements. This is important given that many stakers might leave their tokens in a pool and not pay attention for a long while. However, I agree that potentially allowing operators to have the ability to change fees based on market conditions might be a useful feature in the longer term. But I would class this as a P2